syntax = "proto3";
package qmtl;

import "google/protobuf/timestamp.proto";

// Status/Execution 관련 메시지 (v1.0.0)
// 변경 이력: 최초 생성 2025-05-20

message NodeErrorDetail {
  optional string code = 1;
  optional string message = 2;
  optional string occurred_at = 3;
  optional string recovered_at = 4;
  int32 recovery_count = 5;
  map<string, string> extra = 6;
}

message NodeStatus {
  string node_id = 1;
  string status = 2;
  optional google.protobuf.Timestamp start_time = 3;
  optional google.protobuf.Timestamp end_time = 4;
  map<string, string> result = 5;
  map<string, string> resource = 6;
  map<string, string> meta = 7;
  NodeErrorDetail error_detail = 8;
  optional string last_recovered_at = 9;
  int32 recovery_count = 10;
}

message PipelineStatus {
  string pipeline_id = 1;
  string status = 2;
  map<string, string> params = 3;
  google.protobuf.Timestamp start_time = 4;
  google.protobuf.Timestamp last_update = 5;
  google.protobuf.Timestamp end_time = 6;
  float progress = 7;
  map<string, string> result = 8;
  NodeErrorDetail error_detail = 9;
  optional string last_recovered_at = 10;
  int32 recovery_count = 11;
}

message ExecutionDetail {
  string execution_id = 1;
  string strategy_id = 2;
  string version_id = 3;
  string status = 4;
  optional int64 start_time = 5;
  optional int64 end_time = 6;
  map<string, string> parameters = 7;
  map<string, string> result = 8;
}
